languages/rust/oso/src/repl.rs

```diff
@@ -112,21 +112,20 @@ pub fn main() -> anyhow::Result<()> {
     load_files(&mut oso, &mut args)?;
     loop {
         // get input
-        let mut input: String = match repl.oso_input("query> ") {
+        let input: String = match repl.oso_input("query> ") {
             Ok(input) => input,
             Err(e) => {
                 eprintln!("Readline error: {}", e);
                 break;
             }
         };
 
-        if input.starts_with("%def") {
-            if let Err(e) = oso.load_str(&input[4..]) {
+        if let Some(define) = input.strip_prefix("%def") {
+            if let Err(e) = oso.load_str(define) {
                 println!("{}", e);
             }
             continue;
         }
-        input.pop(); // remove trailing `;`
```

Comment:
(this was incorrect, I went to try out the REPL after making the above change to confirm it works)